utils/pattern/src/frontend/mod.rs

```diff
@@ -13,11 +13,17 @@ use core::{
     marker::PhantomData,
 };
 
+#[cfg(feature = "alloc")]
+use core::str::Utf8Error;
+
 use writeable::{adapters::TryWriteableInfallibleAsWriteable, PartsWrite, TryWriteable, Writeable};
 
 use crate::common::*;
 use crate::Error;
 
+#[cfg(feature = "alloc")]
+use crate::PatternOrUtf8Error;
```

Comment:
Issue: This should work even without `alloc`

Reply:
this is only used in 
```
#[cfg(feature = "alloc")]
impl<B, E> Pattern<B, <B::Store as ToOwned>::Owned>
```

which guarded by feature `alloc`, therefore, we need to guard the use also with feature `alloc` or one of the CI will fail due to warning for unused `use crate::PatternOrUtf8Error`

Reply:
done